private/model/api/shape.go

```diff
@@ -564,6 +598,18 @@ func (ref *ShapeRef) GoTags(toplevel bool, isRequired bool) string {
 	return fmt.Sprintf("`%s`", tags)
 }
 
+// HasPayloadMembers returns if the shape has any members that will be
+// serialized to the payload of a API message.
+func (s *Shape) HasPayloadMembers() bool {
```

Comment:
Does this need to consider whether the input shape is a synthetic shape that wasn't actually modeled as an operation input, but exists purely to maintain a stable API interface?

Reply:
I would expect a synthetic stub shape to to always return false for `HasPayloadMembers`. Since there would be no members under the structure. A Synthetic shape that are clones though may have members that target HTTP bindings, not the payload.

The item that may be missing here, is if there should be any distinction/behavior for members that are NOT top level shapes. Nested shapes serialization is based on the context of the nested shape in the request parameters.